test/ToricVarieties/runtests.jl

```diff
@@ -552,3 +554,78 @@ end
     @test length(intersection_form(dP3)) == 21
     @test integrate(c^2+c-3//4*c*c) == -1//4
 end
+
+
+
+
+
+##################################################
+# (15) Closd subvarieties
+##################################################
+
+(x1, x2, y1, y2) = gens(cox_ring(ntv));
+sv1 = ClosedSubvarietyOfToricVariety(ntv, [x1])
+sv2 = ClosedSubvarietyOfToricVariety(ntv, [x1^2+x1*x2+x2^2,y2])
+sv3 = ClosedSubvarietyOfToricVariety(P3, [gens(cox_ring(P3))[1]^2])
+
+@testset "Test error messages for closed subvarieties" begin
+    @test_throws ArgumentError ClosedSubvarietyOfToricVariety(ntv, [x1 - y1])
+    @test_throws ArgumentError ClosedSubvarietyOfToricVariety(antv6, [gens(cox_ring(antv6))[1]])
+end
+
+@testset "Properties and attributes of closd subvarieties" begin
+    @test is_empty(sv1) == false
+    @test radical(sv1) == defining_ideal(sv1)
+    @test dim(toric_variety(sv1)) == 2
+end
+
+
+
+
+##################################################
+# (16) Algebraic cycles
+##################################################
+
+ac1 = RationalEquivalenceClass(D2)
+ac2 = RationalEquivalenceClass(DC3)
+ac3 = RationalEquivalenceClass(l4)
+ac4 = RationalEquivalenceClass(c3)
+ac5 = RationalEquivalenceClass(CohomologyClass(dP3, x3))
+ac6 = RationalEquivalenceClass(ToricLineBundle(ntv,[1,1]))
+
+@testset "Error of rational equivalence classes" begin
+  @test_throws ArgumentError RationalEquivalenceClass(antv, [1,2,3])
+  @test_throws ArgumentError RationalEquivalenceClass(toric_variety(ac1),[1,2,3])
+  @test_throws ArgumentError ac1 + ac3
+  @test_throws ArgumentError ac1 - ac3
+  @test_throws ArgumentError ac1 * ac3
+  @test_throws ArgumentError sv1 * ac1
+  @test_throws ArgumentError ac2 * sv1
+  @test_throws ArgumentError sv1 * sv3
+end
```

Comment:
@lkastner I believe these tests cover most (if not all?) of the error messages that I added in this PR. Do you agree? If not, which errors need testing?

Reply:
Sorry, I missed these. These are perfect.